### PR TITLE
Task/sl 14 protect routes component

### PR DESCRIPTION
**Pull request by AndriiSokyrko**
task/sl-14-protect-routes-componen

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "lint:type": "npx tsc --noEmit",
     "lint:js": "npx eslint \"src/**/*.ts\"",
     "start:dev": "nodemon --exec \"node --loader ts-paths-esm-loader\" src/index.ts",
-    "migrate:dev": "node --loader ts-paths-esm-loader ../node_modules/knex/bin/cli.js migrate:latest",
+    "migrate:dev": "node --watch --loader ts-paths-esm-loader ../node_modules/knex/bin/cli.js migrate:latest",
```
Comment (andanf-e) [inline backend/package.json]: Don't know for what reason migration script should use `--watch`, and how it relates to ProtectedRoute on FE
```suggestion
    "migrate:dev": "node --loader ts-paths-esm-loader ../node_modules/knex/bin/cli.js migrate:latest",
```

File: frontend/src/index.tsx [hunk 2]
```diff
@@ -12,6 +12,8 @@ import { AppRoute } from '~/libs/enums/enums.js';
 import { store } from '~/libs/packages/store/store.js';
 import { Auth } from '~/pages/auth/auth.js';
 
+import { ProtectRouteComponent } from './pages/auth/components/components.js';
```
Comment (andanf-e) [inline frontend/src/index.tsx]: It should be a common component (placed in libs)

File: frontend/src/index.tsx [hunk 3]
```diff
@@ -23,15 +25,21 @@ createRoot(document.querySelector('#root') as HTMLElement).render(
...
-                element: 'Root',
-              },
-              {
-                path: AppRoute.SIGN_IN,
-                element: <Auth />,
-              },
-              {
-                path: AppRoute.SIGN_UP,
-                element: <Auth />,
+                element: <ProtectRouteComponent />,
```
Comment (andanf-e) [inline frontend/src/index.tsx]: ProtectedRoute should wrap only protected routes (🤔)

File: frontend/src/pages/auth/components/protect-route-component/protect-route-component.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+} from '~/libs/hooks/hooks.js';
+import { storage, StorageKey } from '~/libs/packages/storage/storage.js';
+
+const ProtectRouteComponent = (): React.ReactElement => {
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+  const isAuth = useCallback(async (): Promise<void> => {
+    try {
+      const token = await storage.get(StorageKey.TOKEN);
+      if (!token && pathname !== AppRoute.SIGN_UP) {
```
Comment (andanf-e) [inline frontend/src/pages/auth/components/protect-route-component/protect-route-component.tsx]: ProtectedRoute should not know anything about current route path

File: frontend/src/pages/auth/components/protect-route-component/protect-route-component.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  useEffect,
+  useLocation,
+  useNavigate,
+} from '~/libs/hooks/hooks.js';
+import { storage, StorageKey } from '~/libs/packages/storage/storage.js';
+
+const ProtectRouteComponent = (): React.ReactElement => {
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+  const isAuth = useCallback(async (): Promise<void> => {
```
Comment (andanf-e) [inline frontend/src/pages/auth/components/protect-route-component/protect-route-component.tsx]: Components should not handle asynchronous operations

**Review by andanf-e — CHANGES_REQUESTED** [review]

File: backend/package.json (hunk 1 above)
Comment (AndriiSokyrko) [reply backend/package.json]: "migrate:dev": "node --watch --loader ts-paths-esm-loader ../node_modules/knex/bin/cli.js migrate:latest",

**Pull request closed without merging**